js/data/data_source/data_source.js

```diff
@@ -498,6 +497,10 @@ export const DataSource = Class.inherit({
         const operationId = this._operationManager.add(deferred);
         const storeLoadOptions = this._createStoreLoadOptions();
 
+        if(this._store && !isEmptyObject(storeLoadOptions?.compareOptions)) {
+            this._store._compareOptions = { ...this._store._compareOptions, ...storeLoadOptions.compareOptions };
+        }
```

Comment:
Minor comment.

Nothing scary will happen if we spread the empty object :)

Let's simplify these lines:
```suggestion
        if(this._store) {
            this._store._compareOptions = { ...this._store._compareOptions, ...storeLoadOptions?.compareOptions };
        }
```

Reply:
i think better no reassign store._compareOptions if it's not necessary